### PR TITLE
Add ability to submit to Turnitin after enabling the plugin after submission

### PR DESCRIPTION
**Pull request by dwinn**
Add ability to submit to Turnitin after enabling the plugin after submission

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -505,6 +508,7 @@ public function submission_handler($eventdata) {
         $author = $moduleobject->get_author($eventdata['userid'], $eventdata['relateduserid'], $cm, $eventdata['objectid']);
         $groupid = $moduleobject->get_groupid($eventdata['objectid']);
         $submitter = new plagiarism_turnitinsim_user($eventdata['userid']);
+        // remove this
```
Comment (matdelong) [inline lib.php]: ?

Comment (dwinn) [reply lib.php]: Had some debug lines there, seems I removed the debug but forgot to remove the note.

File: tests/lib_test.php [hunk 2]
```diff
@@ -219,55 +219,6 @@ public function test_get_links_student_view_reports() {
         $this->assertEquals('', $plagiarismturnitinsim->get_links($linkarray));
     }
 
-    /**
-     * Test that get_links returns a not sent message if the submission was not sent.
-     */
-    public function test_get_links_submission_not_sent() {
```
Comment (jmcgettrick) [inline tests/lib_test.php]: This no longer needed?

File: classes/forum.class.php [hunk 3]
```diff
@@ -129,4 +129,48 @@ public function show_other_posts_links($courseid, $userid) {
...
+     */
+    public function create_submission_event_data($linkarray) {
+        global $USER;
+
+        $cm = get_coursemodule_from_id('', $linkarray['cmid']);
+
+        $eventdata = array();
+        $eventdata['contextinstanceid'] = $linkarray['cmid'];
+
+        $eventdata['eventtype'] = 'assessable_submitted';
```
Comment (jmcgettrick) [inline classes/forum.class.php]: Is this event type correct? Is that not only for assign module type?

Comment (dwinn) [reply classes/forum.class.php]: I thought it was the same for all modules but I'll check.

File: tests/lib_test.php (hunk 2 above)
Comment (dwinn) [reply tests/lib_test.php]: That's right. Not sent was used when the plugin was enabled after students had submitted.

File: classes/forum.class.php (hunk 3 above)
Comment (dwinn) [reply classes/forum.class.php]: I can confirm it is the same event type for all three modules. I double checked what is passed through for a submission and it looks like I didn't need to as we define this event type for each one in the observer class anyway.

**Pull request merged**